Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -257,6 +258,79 @@ public async Task<BoolResult> RequestCopyFileAsync(OperationContext context, Con
             }
         }
 
+        /// <summary>
+        /// Pushes content to another machine.
+        /// </summary>
+        public async Task<BoolResult> PushFileAsync(OperationContext context, ContentHash hash, Stream source)
+        {
+            try
+            {
+                var headers = new Metadata()
+                {
+                    { "hash-bin", hash.ToHashByteArray() }, // -bin suffix required to send bytes directly
+                    { "hash_type", hash.HashType.ToString() },
+                    { "trace_id", context.TracingContext.Id.ToString() }
+                };
+
+                var call = _client.PushFile(headers, cancellationToken: context.Token);
+                var requestStream = call.RequestStream;
+
+                var responseHeaders = await call.ResponseHeadersAsync;
+                foreach (var header in responseHeaders)

```

Comment:
Same comment here. Create a type to encapsulate this rather than manipulating headers directly.